lib/python/picongpu/picmi/gaussian_laser.py

```diff
@@ -17,22 +18,63 @@
 class GaussianLaser(picmistandard.PICMI_GaussianLaser):
     """PICMI object for Gaussian Laser"""
 
+    def scalarProduct(self, a: typing.List[float], b: typing.List[float]) \
+        -> float:
+            assert len(a) == len(b), "the scalar product is only defined for two \
+                vector of equal dimension"
+
+            result = 0.
+            for i in range(len(a)):
+                result += a[i] * b[i]
+
+            return result
+
     def __init__(
-            self, wavelength, waist, duration,
+            self, wavelength, waist, duration, propagation_direction,
+            polarization_direction, focal_position, centroid_position,
+            picongpu_polarization_type=
+                laser.GaussianLaser.PolarizationType.LINEAR,
             picongpu_laguerre_modes: typing.Optional[
                 typing.List[float]] = None,
             picongpu_laguerre_phases: typing.Optional[
                 typing.List[float]] = None,
+            picongpu_phase: float = 0.,
+
+            # make sure to always place Huygens-surface inside PML-boundaries,
+            # default is valid for standard PMLs
+
+            # @todo create check for insufficient dimension
+            # @todo create check in simulation for conflict between PMLs and
+            # Huygens-surfaces
+            picongpu_huygens_surface_positions: typing.List[
+                typing.List[int]] = [[16, -16], [16, -16], [16, -16]],
```

Comment:
Do we have in PICME the possibility to define how thick the PML area is? If so these values should be `pml_cells + N` where N>1 (normally we need to know the stencil with of the field solver.

Reply:
No, the PIMCI interface itself only allows setting the following
```
   lower_boundary_conditions: vector of strings
        Conditions at lower boundaries, periodic, open, dirichlet, or neumann
    upper_boundary_conditions: vector of strings
        Conditions at upper boundaries, periodic, open, dirichlet, or neumann
```
and PyPIConGPU does not have a template for FieldAbsorber.param, therefore we rely on the default setting for them. That also means that PyPIConGPU, right now, has no clue what the actual pml settings are unfortunately.
And yes I want to change that, even left a todo in the code, but not now.